sql/modules/Reconciliation.sql

```diff
@@ -93,6 +112,28 @@ $$Sets which lines of the report are cleared.$$;
 CREATE OR REPLACE FUNCTION reconciliation__delete_my_report(in_report_id int)
 RETURNS BOOL AS
 $$
+BEGIN
+    PERFORM approved FROM cr_report
+    WHERE id = in_report_id AND approved;
+    IF FOUND THEN
+        RAISE EXCEPTION 'Cannot delete approved recon report';
+        RETURN FALSE;
+    END IF;
+
+    -- Make sure that transactions present on this report do not have a
+    -- cleared_on date or they won't ever be accessible anymore.
+    PERFORM rl.id
+    FROM cr_report_line rl
+    JOIN acc_trans ac ON rl.ledger_id = ac.entry_id
+    WHERE report_id = in_report_id
+      AND (ac.cleared_on IS NOT NULL OR ac.cleared);
+
+    IF FOUND THEN
+        RAISE EXCEPTION 'Transactions already cleared on this report';
```

Comment:
But... Shouldn't we prevent cleared transactions from being in the report to begin with?